stake-pool/single-pool/src/processor.rs

```diff
@@ -0,0 +1,1079 @@
+            "".to_string(),
+            None,
+            0,
+            true,
+            true,
+            None,
+            None,
+            None,
+        );
+
+        let authority_seeds = &[
+            POOL_AUTHORITY_PREFIX,
+            vote_account_address.as_ref(),
+            &[bump_seed],
+        ];
+        let signers = &[&authority_seeds[..]];
+
+        invoke_signed(
+            &new_metadata_instruction,
+            &[
+                metadata_info.clone(),
+                pool_mint_info.clone(),
+                pool_authority_info.clone(),
+                payer_info.clone(),
+                pool_authority_info.clone(),
+                system_program_info.clone(),
+                mpl_token_metadata_program_info.clone(),
+            ],
+            signers,
+        )?;
+
+        Ok(())
+    }
+
+    #[inline(never)]
+    fn process_update_pool_token_metadata(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        name: String,
+        symbol: String,
+        uri: String,
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let vote_account_info = next_account_info(account_info_iter)?;
+        let pool_authority_info = next_account_info(account_info_iter)?;
+        let authorized_withdrawer_info = next_account_info(account_info_iter)?;
+        let metadata_info = next_account_info(account_info_iter)?;
+        let mpl_token_metadata_program_info = next_account_info(account_info_iter)?;
+
+        check_account_owner(vote_account_info, &vote_program::id())?;
+        let bump_seed = check_pool_authority_address(
+            program_id,
+            vote_account_info.key,
+            pool_authority_info.key,
+        )?;
+        let (pool_mint_address, _) =
+            crate::find_pool_mint_address(program_id, vote_account_info.key);
+        check_mpl_metadata_program(mpl_token_metadata_program_info.key)?;
+        check_mpl_metadata_account_address(metadata_info.key, &pool_mint_address)?;
+
+        // XXX can vote program own other types of accounts? do i need to do any further validation?
+        // XXX has the versions enum *always* existed? ie no accounts exist without it?
+        // i havent seen a V0_23_5 in the wild to confirm the tag bytes are always there
+        // but based on reading the parse routines in monorepo, it *only* parses through the enum
+        let vote_account_data = &vote_account_info.try_borrow_data()?;
+        let state_variant = vote_account_data
+            .get(..VOTE_STATE_START)
+            .and_then(|s| s.try_into().ok())
+            .ok_or(SinglePoolError::UnparseableVoteAccount)?;
+
+        let (withdrawer_start, withdrawer_end) = match u32::from_le_bytes(state_variant) {
+            0 => (LEGACY_VOTE_STATE_START, LEGACY_VOTE_STATE_END),
+            1 => (VOTE_STATE_START, VOTE_STATE_END),
+            _ => return Err(SinglePoolError::UnparseableVoteAccount.into()),
+        };
```

Comment:
The vote program only has vote accounts, so your logic is good. And I don't think there are any more 0.23.5 structs, or if they do exist, they're effectively dead anyway.

I know @t-nelson has been talking about a new vote account version for some time... since it's early in the struct, can we guarantee that the withdrawer key will always be at the same place and make this logic work for new enum variants?

Reply:
we're trying to avoid pulling in the vote program as a dependency here?

withdraw auth is one, one-byte field in front of a variably-length field :sweat_smile:

Reply:
`VoteState::deserialize` cant be used because of stack size violations, it got moved into `solana_program` but had to be ifdeffed out of bpf builds (see https://github.com/solana-labs/solana/issues/30188). this is a workaround until/unless i write a non-serde parser for it